### PR TITLE
Adds docs for creating a cache of rubygems.org

### PR DESCRIPTION
**Pull request by dkliban**
[noissue]

### CODE REVIEW

File: staging_docs/user/guides/04-cache.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
+# Cache rubygems.org 
+
+In this guide you will configure Pulp to act as a pull-through cache of rubygems.org. The first
+time a gem will be requested from Pulp, it will be downloaded from rubygems.org and simultaneously
+streamed to the client. The gem will then be saved in Pulp. On any subsequent requests for the same
+gem, Pulp will return the local copy.
```
Comment (bmbouter) [inline staging_docs/user/guides/04-cache.md:6]:
```suggestion
gem, Pulp will return the saved copy.
```

**Comment by bmbouter** [comment]
Am I right in reading that there is no actual repository created here?

**Comment by dkliban** [comment]
> Am I right in reading that there is no actual repository created here?

That is correct.

File: staging_docs/user/guides/04-cache.md [hunk 2]
```diff
@@ -0,0 +1,114 @@
+# Cache rubygems.org
```
Comment (mdellweg) [inline staging_docs/user/guides/04-cache.md:1]: Is that "build a caching proxy (of rubygems.org)"?

File: staging_docs/user/guides/04-cache.md [hunk 3]
```diff
@@ -0,0 +1,114 @@
+# Cache rubygems.org 
+
+In this guide you will configure Pulp to act as a pull-through cache of rubygems.org. After this,
```
Comment (mdellweg) [inline staging_docs/user/guides/04-cache.md:3]: Please don't do the block formatting. We want to lower the barrier for contributions.

Comment (mdellweg) [inline staging_docs/user/guides/04-cache.md:3]:
```suggestion
In this guide you will configure Pulp to act as a pull-through cache of `rubygems.org`. After this,
```

File: staging_docs/user/guides/04-cache.md [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+the first time a gem will be requested from Pulp, it will be downloaded from rubygems.org and
+simultaneously streamed to the client. The gem will then be saved in Pulp. On any subsequent
```
Comment (mdellweg) [inline staging_docs/user/guides/04-cache.md:5]:
```suggestion
simultaneously streamed to the client. The gem will then be saved in Pulp. On subsequent
```
Let's not be overly specific. This gem is still orphaned...

File: staging_docs/user/guides/04-cache.md (hunk 3 above)
Comment (dkliban) [reply staging_docs/user/guides/04-cache.md:3]: I don't understand what that means. Can you please suggest a specific change?

File: staging_docs/user/guides/04-cache.md (hunk 2 above)
Comment (dkliban) [reply staging_docs/user/guides/04-cache.md:1]: Maybe?

**Review by mdellweg — APPROVED** [review]

**Pull request merged**